src/Sylius/Behat/Context/Ui/CheckoutContext.php

```diff
@@ -156,23 +165,100 @@ public function iSpecifyTheBillingAddressAs(AddressInterface $address)
     }
 
     /**
+     * @When /^I specified the shipping (address as "([^"]+)", "([^"]+)", "([^"]+)", "([^"]+)" for "([^"]+)")$/
+     */
+    public function iSpecifiedTheShippingAddress(AddressInterface $address)
+    {
+        $this->addressingPage->open();
+        $this->addressingPage->specifyShippingAddress($address);
+        $this->iProceedWithTheShippingStep();
+    }
+
+    /**
      * @When I choose the different billing address
      */
     public function iChooseTheDifferentBillingAddress()
     {
         $this->addressingPage->chooseDifferentBillingAddress();
     }
 
+    public function iSpecifyTheBillingSLastNameAs($lastName = null)
+    {
+        $this->addressingPage->specifyBillingAddressLastName($lastName);
+    }
+
+    /**
+     * @When I specify the billing's street as :streetName
+     * @When I do not specify the billing's street
+     */
+    public function iSpecifyTheBillingSStreetAs($streetName = null)
+    {
+        $this->addressingPage->specifyBillingAddressStreet($streetName);
+    }
+
+    /**
+     * @Given I choose :countryName as billing's country
+     */
+    public function iChooseAsBillingSCountry($countryName)
+    {
+        $this->addressingPage->chooseBillingAddressCountry($countryName);
+    }
+
+    /**
+     * @When I specify the billing's city as :cityName
+     * @When I do not specify the billing's city
+     */
+    public function iSpecifyTheBillingSCityAs($cityName = null)
+    {
+        $this->addressingPage->specifyBillingAddressCity($cityName);
+    }
+
     /**
-     * @When I proceed to the next step
-     * @When I try to proceed to the next step
+     * @When I specify the billing's postcode as :postcode
+     * @When I do not specify the billing's postcode
      */
-    public function iProceedToTheNextStep()
+    public function iSpecifyTheBillingSPostcodeAs($postcode = null)
+    {
+        $this->addressingPage->specifyBillingAddressPostcode($postcode);
+    }
+
+    /**
+     * @When I select :shippingMethod shipping method
+     */
+    public function iSelectShippingMethod($shippingMethod)
+    {
+        $this->shippingPage->selectShippingMethod($shippingMethod);
+    }
+
+    /**
+     * @Then I should not be able to select :shippingMethod shipping method
+     */
+    public function iShouldNotBeAbleToSelectShippingMethod($shippingMethod)
+    {
+        Assert::false(
+            $this->shippingPage->hasShippingMethod($shippingMethod),
+            sprintf('Shipping method "%s" should not be available but it does.', $shippingMethod)
+        );
+    }
+
+    /**
+     * @When I proceed to the shipping step
+     * @When I try to proceed to the shipping step
+     */
+    public function iProceedWithTheShippingStep()
     {
         $this->addressingPage->nextStep();
     }
 
     /**
+     * @When I proceed with the payment step
```

Comment:
I see that we need to use appropriate page to confirm the step, but I think it would be nice to decouple the step definitions from what is next. So this could be:

```
I complete the addressing step
I try to complete the addressing step
```

$this->addressingPage->completeStep();

This way we have a bit cleaner structure and if order of steps changes for whatever reason - we can easily reflect that in Behat. WDYT?